src/app/task.service.ts

```diff
@@ -11,7 +11,14 @@ export class TaskService {
 
   constructor() { }
 
+  getId(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
   addTask(task: Task): Observable<Task[]> {
+    task.id = this.getId(1, 500);
```

Comment:
You are not assigning id to the task. As a result all of your tasks dont have id now.

Reply:
![image](https://user-images.githubusercontent.com/45048289/74558250-39015780-4f62-11ea-8085-6a18f544a912.png)
I don't understand, after using a console.lof for list, I see different number of id

Reply:
My bad. You are completely right.